tests/test_core_perim.py

```diff
@@ -0,0 +1,21 @@
+from geomeppy.geom.core_perim import core_perim_zone_coordinates
```

Comment:
What happens if the perimeter depth is greater than half the building depth? We should raise an error in this case, and have a test for it here.

Reply:
It triggers an error in shapely: `NotImplementedError: Multi-part geometries do not provide a coordinate sequence`. How would you like me to deal with it?

I believe that I have addressed the other changes you requested, that's the last item I need to address before committing again.

Reply:
I'd suggest catching that `NotImplementedError` and doing something like `raise ValueError("Perimeter depth is too great")`.

Reply:
Oh, and by all means commit as you go. No need to wait till it's finished.

Reply:
I finally commit and pushed my changes. Let me know what you think! Thanks.